tiledb/api/src/array/mod.rs

```diff
@@ -380,6 +381,97 @@ impl Array {
         Ok(Some(datatype))
     }
 
+    /// Cleans up the array, such as consolidated fragments and array metadata.
+    pub fn vacuum<S>(
+        ctx: &Context,
+        array_uri: S,
+        config: &Config,
+    ) -> TileDBResult<()>
+    where
+        S: AsRef<str>,
+    {
+        let c_array_uri = cstring!(array_uri.as_ref());
+        ctx.capi_call(|ctx| unsafe {
+            ffi::tiledb_array_vacuum(ctx, c_array_uri.as_ptr(), config.capi())
+        })?;
+        Ok(())
+    }
+
+    /// Upgrades an array to the latest format version.
+    pub fn upgrade_version<S>(
+        ctx: &Context,
+        array_uri: S,
+        config: &Config,
```

Comment:
Hm, maybe this should be `Option` as it sounds from the C API description that it is not required.

Reply:
I've checked the c api and looks like it accepts null, so I will update. Same as below.